Documented DebOps fold format as example and fully support it in yaml2rst.

ypid opened the pull request:
@htgoebel Good thing that you waited with the release. There was another extention needed because of a little [redesign](https://github.com/debops/docs/issues/156#issuecomment-231670793).

Docs: https://github.com/ypid/yaml2rst/blob/feature/yaml-fold-strip/docs/fold-markers.rst

Related to: #3
Related to: https://github.com/debops/docs/issues/156

ypid committed:
Use full lenght underline including the fold markers.

Ref: https://github.com/debops/docs/issues/156#issuecomment-231672591

One thing that @drybjed changed was to go from

  ```YAML
  # Main configuration [[[
  # ------------------
  ```

to:

  ```YAML
  # Main configuration [[[
  # ----------------------
  ```

This has the advantage that in case we ever manage to get syntax highlighting for the embedded RST syntax working in Vim, it will not complain about sections being to short. I tried it but I am not sure if that is possible as the `# ` in the input for the syntax highlighting in Vim would need to be stipped out and then passed to the `rst` syntax highlighting. Ref: http://vim.wikia.com/wiki/Different_syntax_highlighting_within_regions_of_a_file#Extended_version

ypid committed:
Fold format: Documented decision process and outcome for the DebOps project.

Related to: https://github.com/debops/docs/issues/156

ypid committed:
Implemented --yaml-strip-regex to strip out things in YAML state.

`--strip-regex` was intentionally limited to the RST part of the input
to avoid problems.
`--yaml-strip-regex` allows to mess with the YAML part of the input if
needed.

Related to: #3
Related to: https://github.com/debops/docs/issues/156

ypid committed:
Unit tests don’t pass strip params to convert. Default to None.

htgoebel commented on Makefile:
```diff
@@ -53,7 +53,9 @@ dist:
 	PYTHONPATH=.
-	bin/yaml2rst examples/main.yml examples/main.rst --strip-regex '\s*(:?\[{3}|\]{3})\d?$$'
+	sed --regexp-extended 's/(\.\. )envvar(::)/\1note\2/;' examples/fold-markers-debops.yml examples/main.yml \
+		| bin/yaml2rst - examples/main.rst --strip-regex '\s*(:?\[{3}|\]{3})\d?$$' --yaml-strip-regex '^\s{66,67}#\s\]{3}\d?$$'
```
Huh, This is very hard to read. I suggest using an intermediate target (is it `examples/main.yml`?)

ypid replied:
Good idea. Done. You might need to use `make examples -B` now as `make` does not check all the dependencies of `yaml2rst` (e. g. when you make a change in the Python module).

ypid committed:
Use intermediate make targets in the Makefile.

Related to: https://github.com/htgoebel/yaml2rst/pull/6/files/f2828f0f04632565b32368b741a16109c28501e1#r71327173

htgoebel commented:
Thanks for the quick reaction.
